### DISCUSSION TITLE
Migrate to Postgress DB

### DISCUSSION THREAD

**Discussion by rui0707** [inventree/InvenTree#5399] [Q&A]
Hi All, 
I am hoping someone could help me out, I have installed inventree using the install script provided, everything works really well.
When I try to change the database from SQLLite3 to postgress I ran into problems, the migration appears to run successfully but after rebooting I get a 502 error when reloading the webpage.
I have tried all of the steps in the instructions, have tried the docker install, bare metal and always get the same outcome.

Hope someone can share their experience in migrating from one db to another.

Thanks

**Comment by SchrodingersGat** [comment]
> the migration appears to run successfully but after rebooting I get a 502 error when reloading the webpage.

If the migrations run OK, then the transfer process has worked! A 502 error would indicate that there is something wrong with either your InvenTree server or the proxy server you have in front of it. Please provide:

- Version information of InvenTree
- Install method (docker / installer / bare metal / etc)
- Are you running in DEBUG mode?
- More details on the 502 message
- Any logs you can provide

**Comment by rui0707** [comment]
Hi,
I have done a bare metal install, running version 0.12.1, since I posted I re-installed using the script provided in the install section, bare metal and I am trying to follow the instructions to migrate db.
What I do next is to change the config.yaml file in the opt/inventree/InvenTree directory to use my postgress db after which I run the invoke update cmd from the /opt/inventree location and 
Not sure where I can find the logs, if you can assist in finding them I can then post them here.

Inventree Config:

root@ip-172-31-0-136:/opt/inventree# inventree config
APP_GROUP=inventree
APP_HOME=/opt/inventree
APP_NAME=inventree
APP_RUNNER_CLI=systemctl
APP_RUNNER_TYPE=systemd
APP_SAFE_NAME=inventree
APP_USER=inventree
APP_WIZARDS=
FORWARDED_ALLOW_IPS=*
GUNICORN_CMD_ARGS=--access-logfile -
HOME=/home/inventree
INVENTREE_CONFIG_FILE=/etc/inventree/config.yaml
LANG=en_US.UTF-8
OLDPWD=/
ORIGINAL_PWD=/opt/inventree
PATH=/opt/inventree/.heroku/python/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/sbin
PORT=6000
PWD=/opt/inventree
PYTHONHASHSEED=random
PYTHONPATH=/opt/inventree
PYTHONUNBUFFERED=true
ROOT_PATH=/
SHLVL=0

**Comment by rui0707** [comment]
I ran the invoke update cmd and this is a snippet of an error that popped up.

load version from file
[PLUGIN] Encountered an error with django:
App registry isn't ready yet.
Operations to perform:
  Synchronize unmigrated apps: allauth, allauth_2fa, corsheaders, crispy_forms, dbbackup, dj_rest_auth, django_cleanup, django_filters, django_ical, django_otp, djmoney, drf_spectacular, formtools, generic, import_export, maintenance_mode, markdownify, messages, mptt, registration, rest_framework, sslserver, staticfiles
  Apply all migrations: InvenTree, account, admin, auth, authtoken, build, common, company, contenttypes, django_q, error_report, exchange, flags, label, order, otp_static, otp_totp, part, plugin, report, sites, socialaccount, stock, taggit, user_sessions, users
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  No migrations to apply.
load version from file
System check identified no issues (1 silenced).
========================================
InvenTree database migrations completed!

**Comment by SchrodingersGat** [comment]
@matmair is this another issue related to the installer cli?

**Reply by matmair** [reply]
@SchrodingersGat no, the user is changing an unrelated file - as you can see in the config section (and as is customary with debian/ubuntu) the config file is ins /etc/inventree/.

**Reply by rui0707** [reply]
So to change Inventree to migrate to Postgres is to change the config file in /etc/inventree and run the invoke migrate cmd?

**Reply by SchrodingersGat** [reply]
@rui0707 you will first need to ensure that you have exported your data *before* performing the changes to your setup!

**Reply by rui0707** [reply]
yes I have backed it up, also I haven't gone into production yet, the issue is I can't get it to migrate to postgress db.

**Comment by rui0707** [comment]
Hi Guys,
Any ideas on how I should try to migrate to postgress, any additional steps?
Should i run the invoke update from /opt/inventree after changing the config.yaml file from /opt/inventree/InvenTree folder?
Do I need to manually change the config.yaml in the /etc/inventree to use postgress? When I did this before it causes the 502 error and I need to revert back to using SQL3lite.

Thanks

**Comment by rui0707** [comment]
Hi All,
Can you advise the correct process to migrate to Postgress if I am modifying the incorrect file?

Get Outlook for Android<https://aka.ms/AAb9ysg>

Rui Clemente - IT Support Manager
76 Heathcote Rd, Moorebank 
NSW 
2170
P: 02 9758 9555 | M: 0401 129 399
FAQs - 
http://www.gegroup.com.au
The information in this email is confidential.  If you are not the intended recipient, you must not distribute, copy, disclose or use the information or attached files in this email in any way.  We do not guarantee that the integrity of this communication has been maintained nor that it is free of errors or viruses.
________________________________
From: Matthias Mair ***@***.***>
Sent: Sunday, August 13, 2023 3:57:45 AM
To: inventree/InvenTree ***@***.***>
Cc: Rui Clemente ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [inventree/InvenTree] Migrate to Postgress DB (Discussion #5399)


@SchrodingersGat<https://github.com/SchrodingersGat> no, the user is changing an unrelated file - as you can see in the config section (and as is customary with debian/ubuntu) the config file is ins /etc/inventree/.
